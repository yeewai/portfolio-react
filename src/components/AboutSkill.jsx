import React from 'react';
import { Tooltip } from 'react-tippy';

const AboutSkill = ({title, shortName, skillLevel}) => (
    <li class={`skill-${skillLevel}`}>
        <Tooltip arrow={true} title={title}>{shortName}</Tooltip>
    </li>
)

export default AboutSkill;
