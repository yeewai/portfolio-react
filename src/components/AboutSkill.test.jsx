import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import AboutSkill from './AboutSkill';

Enzyme.configure({ adapter: new Adapter() });

describe("(About) Skill", () => {
    it('renders', () => {
      const wrapper = shallow(<AboutSkill title="Such a cool title" shortName="sup" skillLevel="full"/>);
      expect(wrapper).toMatchSnapshot();
    });
});
