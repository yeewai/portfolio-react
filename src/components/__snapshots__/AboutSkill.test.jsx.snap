// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(About) Skill renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AboutSkill
    shortName="sup"
    skillLevel="full"
    title="Such a cool title"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Tooltip
        animateFill={true}
        animation="shift"
        arrow={true}
        arrowSize="regular"
        className=""
        delay={0}
        disabled={false}
        distance={10}
        duration={375}
        followCursor={false}
        hideDelay={0}
        hideDuration={375}
        hideOnClick={true}
        html={null}
        inertia={false}
        interactive={false}
        interactiveBorder={2}
        multiple={false}
        offset={0}
        onHidden={[Function]}
        onHide={[Function]}
        onRequestClose={[Function]}
        onShow={[Function]}
        onShown={[Function]}
        popperOptions={Object {}}
        position="top"
        size="regular"
        sticky={false}
        stickyDuration={200}
        style={Object {}}
        theme="dark"
        title="Such a cool title"
        touchHold={false}
        trigger="mouseenter focus"
        unmountHTMLWhenHide={false}
>
        sup
</Tooltip>,
      "class": "skill-full",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "animateFill": true,
        "animation": "shift",
        "arrow": true,
        "arrowSize": "regular",
        "children": "sup",
        "className": "",
        "delay": 0,
        "disabled": false,
        "distance": 10,
        "duration": 375,
        "followCursor": false,
        "hideDelay": 0,
        "hideDuration": 375,
        "hideOnClick": true,
        "html": null,
        "inertia": false,
        "interactive": false,
        "interactiveBorder": 2,
        "multiple": false,
        "offset": 0,
        "onHidden": [Function],
        "onHide": [Function],
        "onRequestClose": [Function],
        "onShow": [Function],
        "onShown": [Function],
        "popperOptions": Object {},
        "position": "top",
        "size": "regular",
        "sticky": false,
        "stickyDuration": 200,
        "style": Object {},
        "theme": "dark",
        "title": "Such a cool title",
        "touchHold": false,
        "trigger": "mouseenter focus",
        "unmountHTMLWhenHide": false,
      },
      "ref": null,
      "rendered": "sup",
      "type": [Function],
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Tooltip
          animateFill={true}
          animation="shift"
          arrow={true}
          arrowSize="regular"
          className=""
          delay={0}
          disabled={false}
          distance={10}
          duration={375}
          followCursor={false}
          hideDelay={0}
          hideDuration={375}
          hideOnClick={true}
          html={null}
          inertia={false}
          interactive={false}
          interactiveBorder={2}
          multiple={false}
          offset={0}
          onHidden={[Function]}
          onHide={[Function]}
          onRequestClose={[Function]}
          onShow={[Function]}
          onShown={[Function]}
          popperOptions={Object {}}
          position="top"
          size="regular"
          sticky={false}
          stickyDuration={200}
          style={Object {}}
          theme="dark"
          title="Such a cool title"
          touchHold={false}
          trigger="mouseenter focus"
          unmountHTMLWhenHide={false}
>
          sup
</Tooltip>,
        "class": "skill-full",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "animateFill": true,
          "animation": "shift",
          "arrow": true,
          "arrowSize": "regular",
          "children": "sup",
          "className": "",
          "delay": 0,
          "disabled": false,
          "distance": 10,
          "duration": 375,
          "followCursor": false,
          "hideDelay": 0,
          "hideDuration": 375,
          "hideOnClick": true,
          "html": null,
          "inertia": false,
          "interactive": false,
          "interactiveBorder": 2,
          "multiple": false,
          "offset": 0,
          "onHidden": [Function],
          "onHide": [Function],
          "onRequestClose": [Function],
          "onShow": [Function],
          "onShown": [Function],
          "popperOptions": Object {},
          "position": "top",
          "size": "regular",
          "sticky": false,
          "stickyDuration": 200,
          "style": Object {},
          "theme": "dark",
          "title": "Such a cool title",
          "touchHold": false,
          "trigger": "mouseenter focus",
          "unmountHTMLWhenHide": false,
        },
        "ref": null,
        "rendered": "sup",
        "type": [Function],
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
